### PR TITLE
refactor mobile push to avoid a type of error CORE-8817

### PR DESCRIPTION
**Pull request by mmaxim**
If the user get an audible push notification from an edit (because an @ message of them was edited into it), then we can get a black bar if that notification is clicked. The reason is that the logic for unboxing a message on an audible notification for caching purposes still attempts to compute the local notification text for a silent notification. That fails for an edit, and the user is presented with a black bar on Android.

### CODE REVIEW

File: go/chat/mobilepush.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	return &MobilePush{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "MobilePush", false),
+	}
+}
+
+func (h *MobilePush) AckNotificationSuccess(ctx context.Context, pushIDs []string) {
+	defer h.Trace(ctx, func() error { return nil }, "AckNotificationSuccess")()
+	// Get session token
+	nist, _, err := h.G().ActiveDevice.NISTAndUID(ctx)
```
Comment (joshblum) [inline go/chat/mobilepush.go]: can this get conn/nist situation get wrapped up in a helper? didn't we need this in the extension as well?

File: go/chat/mobilepush.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	defer conn.Shutdown()
+
+	// Make remote successful call on our ad hoc conn
+	cli := chat1.RemoteClient{Cli: NewRemoteClient(h.G(), conn.GetClient())}
+	if err = cli.RemoteNotificationSuccessful(ctx,
+		chat1.RemoteNotificationSuccessfulArg{
+			AuthToken:        gregor1.SessionToken(nist.Token().String()),
+			CompanionPushIDs: pushIDs,
+		}); err != nil {
+		h.Debug(ctx, "AckNotificationSuccess: failed to invoke remote notification success: %",
```
Comment (joshblum) [inline go/chat/mobilepush.go]: `%v", err)`

Comment (mmaxim) [reply go/chat/mobilepush.go]: I think this is fixed in my extension PR, its been like this forever,

File: go/chat/mobilepush.go (hunk 1 above)
Comment (mmaxim) [reply go/chat/mobilepush.go]: Sure

**Comment by mmaxim** [comment]
@joshblum PTAL

**Pull request merged**